src/libmongoc/tests/test-libmongoc.c

```diff
@@ -2094,17 +2094,18 @@ _parse_server_version (const bson_t *buildinfo)
    ASSERT (bson_iter_recurse (&iter, &array_iter));
 
    /* Server returns a 4-part version like [3, 2, 0, 0], or like [3, 2, 0, -49]
-    * for an RC. Bail if number of parts is ever not 4. */
+    * for an RC. Ignore the 4th part since RCs are equivalent to non-RCs for
+    * testing purposes. */
    i = 0;
    while (bson_iter_next (&array_iter)) {
+      if (++i == N_SERVER_VERSION_PARTS) {
```

Comment:
Why not change `N_SERVER_VERSION_PARTS` to 3. I think that would ensure we're only looking at the first 3 parts though I haven't tested this locally so it may be something I'm missing.

Reply:
Done. It's about the same amount of code, as you still need to break early when iterating through server versions, but it seems more accurate to change `N_SERVER_VERSION_PARTS` to 3.